File: API/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;

namespace API.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        public long DeliveryFees { get; set; }
        public ShippingAddress ShippingAddress { get; set; }
        public string PaymentIntentId { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.isPending;
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public long Subtotal { get; set; }

        public long GetTotal() {
           return DeliveryFees + Subtotal;
        }
    }
   
}

File: client/src/app/models/Order.ts
export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    country: string;
    zipCode: string;
}

export interface OrderItem {
    productId: number;
    name: string;
    pictureUrl: string;
    description: string;
    price: number;
    quantity: number;
    brand?: string;
    typeofProduct?: string;
}

export interface Order {
    id: number;
    buyerId: string;
    deliveryFees: number;
    shippingAddress: ShippingAddress;
    orderItems: OrderItem[];
    orderStatus: string;
    orderDate: string;
    subtotal: number;
    total: number;
}